src/Lumi/Components2/Banner.purs

```diff
@@ -0,0 +1,223 @@
+                                          $ S._alignSelf S.Start
+                                          $ S.onDesktop (S._alignSelf S.Center)
+                                          $ style_
+                                              ( css
+                                                  { marginRight: int 16
+                                                  }
+                                              )
+                                          $ _ { content = [ icon ]
+                                              }
+                                        _ -> mempty
+                                    ]
+                                    <> props.content
+                                }
+                          ]
+                      }
+                , Monoid.guard props.dismissable
+                    $ button
+                    $ S._alignSelf S.Start
+                    $ dismissButtonStyle
+                    $ _ { content =
+                            [ Icon.icon
+                                { style: R.css { fontSize: "12px" }
+                                , type_: Icon.Remove
+                                }
+                            ]
+                        , onPress = setVisible \_ -> false
+                        }
+                ]
+            }
+  where
+    defaults :: Record BannerProps
+    defaults =
+      { component: Banner
+      , title: Nothing
+      , content: []
+      , dismissable: false
+      , icon: Nothing
+      }
+
+    dismissButtonStyle :: forall props. PropsModifier props
+    dismissButtonStyle =
+      S._interactive
+      <<< style \(LumiTheme { colors }) ->
+            css
+              { ariaLabel: str "dismiss"
+              , outline: str "none"
+              , border: str "none"
+              , background: str "none"
+              , color: color colors.black1
+              , minWidth: str "initial"
+              , padding: int 0
+              , "@media (min-width: 860px)": nested $ css
+                  { padding: int 0
+                  }
+              , "&:hover": nested $ css
+                  { color: color colors.black1
+                  , border: str "none"
+                  , background: str "none"
+                  }
+              , "&:focus": nested $ css
+                  { boxShadow: str "none"
+                  }
+              }
+
+actionBanner :: Array JSX -> PropsModifier BannerProps
+actionBanner actions f =
+  f >>> \props -> props
+    { content =
+        [ box
+          $ S._column
+          $ S.onDesktop (S._row)
+          $ S._flex
+          $ S._align S.End
+          $ style_
+              ( desktopQuery $ css
+                  -- @NOTE positioning of the action button changes dependent on if there is a banner title or not
+                  { alignItems: if (isJust props.title)
+                      then str "flex-end"
+                      else str "center"
```

Comment:
I think this style should go in the `box` that contains only the actions, otherwise the banner content also gets right-aligned on mobile:
![Screen Shot 2020-05-12 at 09 59 30](https://user-images.githubusercontent.com/2164548/81695380-ea683a00-9438-11ea-8eba-d79b16af0d5c.png)

Reply:
Also, from looking at the design documentation, it seems that it's the presence of an _icon_ that affects the positioning of the action buttons. Anyway, I still think we don't need this special casing though, as it looks weird also from a UX perspective to have the buttons change side when an icon exists; that pattern change might confuse users. I'm discussing that with Bethany in the design document.